firmware/projects/TMS/inc/app.h

```diff
@@ -151,4 +153,45 @@ class DebugIndicator {
     inline void UpdateDO() {
         digital_output_.Set(state_);
     }
-};
\ No newline at end of file
+};
+
+class BmsBroadcaster {
+public:
+    BmsBroadcaster(shared::can::CanBus& can_bus, uint8_t num_thermistors)
+        : can_bus_(can_bus), num_thermistors_(num_thermistors) {}
+
+    void SendBmsBroadcast(uint8_t high_thermistor_id,
+                          int8_t high_thermistor_value,
+                          uint8_t low_thermistor_id,
+                          int8_t low_thermistor_value,
+                          int8_t avg_thermistor_value) {
+        bms_broadcast_.high_therm_id = high_thermistor_id;
+        bms_broadcast_.high_therm_value = high_thermistor_value;
+        bms_broadcast_.low_therm_id = low_thermistor_id;
+        bms_broadcast_.low_therm_value = low_thermistor_value;
+        bms_broadcast_.avg_therm_value = avg_thermistor_value;
+        bms_broadcast_.num_therm_en = num_thermistors_;
+        bms_broadcast_.therm_module_num = kThermistorModuleNumber;
+        bms_broadcast_.checksum = CalculateBmsBroadcastChecksum(bms_broadcast_);
+
+        can_bus_.Send(bms_broadcast_);
+    }
+
+private:
+    static constexpr uint8_t kThermistorModuleNumber = 0;
+
+    shared::can::CanBus& can_bus_;
+    generated::can::BmsBroadcast bms_broadcast_;
+    uint8_t num_thermistors_;
```

Comment:
Why does `num_thermistors` need to be an instance variable? It seems like it would fit in better as an argument to `SendBmsBroadcast()`. I know this means that we repeatedly pass the value, but it reduces coupling at construction time, and at call time we will already be handling the temperature arrays so the number will be readily available.

Reply:
Made it an instance because we know the number of thermistors at construction time and it will not change over the course of its lifetime. I think this coupling makes sense, thoughts?

Reply:
Sure it's fine either way. Template it if you can (like the TMS temp manager) but don't if it gets complicated